docker/docker.go

```diff
@@ -136,3 +133,13 @@ func getSwarmConfiguration(config *agent.RuntimeConfiguration, dockerInfo types.
 
 	return nil
 }
+
+func getCli() (*client.Client, error) {
```

Comment:
Maybe this function could take a callback `func(cli *client.Client)` so that the other functions don't need to handle the errors nor the resources.

Something like this:
```go
func ContainerStart(name string, opts types.ContainerStartOptions) error {
	return withCli(func(cli *client.Client) error {
		return cli.ContainerStart(context.Background(), name, opts)
	})
}
```